services/app-api/src/resolvers/submitHealthPlanPackage.ts

```diff
@@ -118,14 +138,27 @@ function submit(
 export function submitHealthPlanPackageResolver(
     store: Store,
     emailer: Emailer,
-    emailParameterStore: EmailParameterStore
+    emailParameterStore: EmailParameterStore,
+    launchDarkly: LDService
 ): MutationResolvers['submitHealthPlanPackage'] {
     return async (_parent, { input }, context) => {
         const { user, span } = context
         const { submittedReason, pkgID } = input
         setResolverDetailsOnActiveSpan('submitHealthPlanPackage', user, span)
         span?.setAttribute('mcreview.package_id', pkgID)
 
+        const rateCertAssuranceFlag = await launchDarkly.getFeatureFlag(
+            user,
+            'rate-cert-assurance'
+        )
+
+        if (rateCertAssuranceFlag instanceof Error) {
+            const msg = `error retrieving feature flag rate-cert-assurance: ${rateCertAssuranceFlag.message}`
+            logError('submitHealthPlanPackage', msg)
+            setErrorAttributesOnActiveSpan(msg, span)
+            throw new ApolloError(msg, 'LD_FETCH_ERROR', { message: msg })
```

Comment:
I dont think LD being down should cause all new submissions to fail. I would think we rely on the SDK fallback failure for that flag here instead. Can you share more about the case where this was needed

Reply:
In this  case we are checking for `riskBasedContract` in `submitHealthPlanPackage` only if the flag `rate-cert-assurance` is on (`true`). This prevents submissions from being submitted when `riskBasedContract` is undefined and the flag is on and allows submissions to be submitted without this check when the flag is off (false).

The reason we need to do this is because an old submission that gets unlocked should not be able to be resubmitted when the question `Is this a risk-based contract?` is unanswered, which is only a requirement when the flag is on.

My thought for the failure here is to keep incomplete submissions from being submitted if LaunchDarkly fails to fetch the flag. The SDK fallback value is also set to undefined, because if set to `true` or `false` one of the two states of the app (flag on or off) would be broken.

Reply:
Taking this to slack for some more discussion. Thanks for bringing up this topic.